_posts/2018/2018-01-26-optimizing-webpack.md

```diff
@@ -0,0 +1,221 @@
+---
+layout: post
+title: Optimizing Webpack for Production
+excerpt: Experimenting with webpack in production
+author:
+  name: Agrim Mittal
+  link: https://agrim123.github.io
+  bio: Developer, SDSLabs
+  image: agrim.jpg
+---
+
+I had been involved in building the official website of [Cognizance](https://cognizance.org.in), The Annual Technical Festival of IIT Roorkee. It is also Asia's second largest tech fest.
+
+### Brief Background
+
+The website is built on [ReactJS](https://reactjs.org/) and [Node.js](https://nodejs.org) It is a typical end to end application.
+
+I will focus on the react part of the application which is one of the client of our Node API.
+
+### n00b steps
+
+Now, as you might know, when webpack does it’s **magic** on the code, it generates a single bundle file, that contains all the code needed in the front end and that can be really huge. Because we are working with single page applications here, this means that browser will fetch all code at once which can lead to slower page loads which can be a bad experience for end users.
```

Comment:
`when webpack does its* maic`

Reply:
`it generates a huge bundle file, that contains all the code required for the front-end to function`